### PR TITLE
Bug 1948719: update dependencies to 1.21

### PR DESCRIPTION
**Pull request by alexander-demichev**
Update dependencies to 1.21

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -203,7 +203,8 @@ func newConfigManagedClient(mgr manager.Manager) (runtimeclient.Client, manager.
 		Scheme: mgr.GetScheme(),
 		Mapper: mgr.GetRESTMapper(),
 	}
-	client, err := manager.NewClientBuilder().Build(cache, mgr.GetConfig(), clientOpts)
+
+	client, err := runtimeclient.New(config.GetConfigOrDie(), clientOpts)
```
Comment (JoelSpeed) [inline cmd/manager/main.go:207]: This client will be uncached, where the previous client is cached, I'm not sure this is correct

Comment (JoelSpeed) [reply cmd/manager/main.go:207]: Perhaps you need to use a `client.NewDelegatingClient` instead?

**Comment by elmiko** [comment]
/title Bug 1948719: update dependencies to 1.21

**Comment by elmiko** [comment]
/retitle Bug 1948719: update dependencies to 1.21

File: cmd/manager/main.go (hunk 1 above)
Comment (alexander-demichev) [reply cmd/manager/main.go:207]: like this?
```
	client, err := runtimeclient.New(config.GetConfigOrDie(), clientOpts)
	if err != nil {
		return nil, nil, err
	}

	delegatedClient, err := runtimeclient.NewDelegatingClient(runtimeclient.NewDelegatingClientInput{
		Client:      client,
		CacheReader: cache,
	})
```

Comment (JoelSpeed) [reply cmd/manager/main.go:207]: Yep, that's the one, that's basically what https://github.com/kubernetes-sigs/controller-runtime/blob/7ef2da0bc161d823f084ad21ff5f9c9bd6b0cc39/pkg/cluster/cluster.go#L259 does, perhaps could use it instead?

**Pull request merged**